### PR TITLE
Fix build and update examples

### PR DESCRIPTION
**Pull request by andrew-lei**
`li` needed to `Output` in its type before it could build.

I've updated the examples to the latest version.

### CODE REVIEW

File: examples/counters/src/Version2.purs [hunk 1]
```diff
@@ -27,9 +28,9 @@ counterModel { increment, decrement, delete } id = do
   count <- sample $ scan (+) 0 changes
   pure { count, delete: delete $> id }
 
-counterView :: CounterOut -> Component CounterViewOut CounterViewOut
+--counterView :: CounterOut -> Unit -> Component CounterViewOut CounterViewOut
```
Comment (paldepind) [inline examples/counters/src/Version2.purs]: Was commenting this line intentional?

File: examples/counters/src/Version2.purs [hunk 2]
```diff
@@ -62,7 +63,7 @@ counterListModel { addCounter, listOut } init = do
   counterIds <- sample $ scan ($) init (appendCounter <> removeCounter)
   pure { sum, counterIds }
 
-counterListView :: ListOut -> Component _ ListViewOut
+--counterListView :: ListOut -> Array Int -> Component _ ListViewOut
```
Comment (paldepind) [inline examples/counters/src/Version2.purs]: This line is also commented.

**Comment by andrew-lei** [comment]
No, those are unintentional. I commented those out to check if the type annotation was preventing it from type checking correctly, but I think they're the correct type. I can check in a bit.

**Comment by paldepind** [comment]
Thanks a lot @andrew-lei. The PR looks really good 👍  All the examples are correctly adapted to the changes in Turbine and Hareactive.

There's just two type annotations that have been commented out? Was that intentional?

May I ask how you found purescript-turbine? Please ask any questions that you may have with regards to the framework. I'm currently working on finishing the TodoMVC example and a small personal project based on purescript-turbine. Once they're done and issues have been ironed out I plan to write some documentation.

**Comment by andrew-lei** [comment]
I was looking up PureScript frameworks, and this was on a list that was the first Google result for that: https://github.com/passy/awesome-purescript

**Comment by paldepind** [comment]
Ah. I didn't no Turbine was on that list :smile: But, the project isn't really ready for general usage yet. It'll probably get there in a couple of months.

**Comment by andrew-lei** [comment]
I think it looks complete enough for my purposes. I was wanting to try and do the 'Excel 365 in 60 minutes' talk (which was in F#/Fable), and I think all the ingredients are there. I'll let you know how it goes.

**Comment by andrew-lei** [comment]
OK those have now been uncommented.

**Comment by paldepind** [comment]
The biggest thing missing right now is probably routing. But yes, all the fundamentals are in place.

> I'll let you know how it goes.

Definitely, do that. And let me know if you have any questions!

**Pull request merged**